### ISSUE TITLE
Escaped records are always updated

### ISSUE THREAD

**Issue by bleetube** [octodns/octodns-easydns#11]
TXT records that are escaped update on every subsequent run `octo-sync`. You won't see this if you manually create the record on the Easydns webui and then `octo-dump`. It happens when I use `octo-sync` to create or update a record, and use any escapes.

A single escape:
```yaml
_dmarc:
  type: TXT
  value: v=DMARC1\; p=none
```
After `octo-sync --doit`,  octo-sync still wants to update it again:
```
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['"v=DMARC1\; p=none"']> ->
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['v=DMARC1\; p=none']> (config)
```
Here is the resulting `octo-dump`:
```yaml
_dmarc:
  type: TXT
  value: '"v=DMARC1\; p=none"'
```
With that in my configuration file, `octo-sync` still wants to update it again:
```
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['"v=DMARC1\; p=none"']> ->
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['v=DMARC1\; p=none']> (config)
```

Other escaping variations still get the same behavior:
```yaml
  value: 'v=DMARC1\; p=none'
```
After `octo-sync --doit`,  octo-sync still wants to update it again:
```
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['"v=DMARC1\; p=none"']> ->
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['v=DMARC1\; p=none']> (config)
```

I saw a [similar issue](https://github.com/octodns/octodns/issues/52) discussed some time ago, where the author describes double escaping _twice_. That still doesn't work:
```yaml
  value: '"v=DMARC1\\; p=none"'
```
After `octo-sync --doit`,  octo-sync still wants to update it again:
```
*   Update
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['"v=DMARC1\\; p=none"']> ->
*     <TxtRecord TXT 3600, _dmarc.bitcoiner.social., ['v=DMARC1\\; p=none']> (config)
```

Versions:
- octodns                   0.9.21
- octodns-easydns           0.0.1
- PyYAML                    6.0

Config:
```yaml
---
manager:
  include_meta: False
  max_workers: 1

providers:
  config:
    class: octodns.provider.yaml.YamlProvider
    directory: ./config
    default_ttl: 3600
    enforce_order: False
  easydns:
    class: octodns_easydns.EasyDnsProvider
    token: env/EASYDNS_TOKEN
    api_key: env/EASYDNS_KEY

zones:
  bitcoiner.social.:
    sources:
      - config
    targets:
      - easydns
```
Basically it seems like the easydns api create/updates are forcing quotes. If you have double quotes, it then escapes your quotes with single quotes. If you have single quotes on your double quotes, it strips the singles. And so on.

**Comment by ross** [comment]
Hrm. Likely an issue with escaping/quoting in what's sent up to the API. I'll take a look at the code when I get a chance, but I don't believe I have access to EasyDNS atm so that may complicate debugging/fixing things.

**Comment by ross** [comment]
Tried signing up for an account and enabling the API for it. That seems to work, but I don't appear to be able to create domains, either with the provider or through their control panel without paying exorbitant $$$. Someone else who has access will have to dig into this.
